### PR TITLE
feat: Creates s3 credentials and bucket automatically

### PR DESCRIPTION
**Pull request by jjaferson**
Automates the creation of s3 credentials and bucket 

https://issues.jboss.org/browse/INTLY-4009

## Verification steps

1. Run a full install of the integreatly-operator:
`operator-sdk up local --namespace=<integreatly_operator_namespace>`

2. Verify if the `s3-credentials` and `s3-bucket` secrets are created:
`oc get secret`

3. Ensure installation of all products has been completed successfully

### CODE REVIEW

**Comment by aidenkeating** [comment]
@jjaferson What's the reasoning here for not leveraging the Cloud Resource Operator here to create the bucket and provide the credentials, similar to https://github.com/integr8ly/integreatly-operator/pull/206 ?

Of course, we're still blocked on OSD, but the Cloud Resource Operator will create a placeholder secret at the moment until OSD has been unblocked.

**Comment by jjaferson** [comment]
> @jjaferson What's the reasoning here for not leveraging the Cloud Resource Operator here to create the bucket and provide the credentials, similar to #206 ?
> 
> Of course, we're still blocked on OSD, but the Cloud Resource Operator will create a placeholder secret at the moment until OSD has been unblocked.


That makes more sense indeed, I will make the changes and update the PR

**Comment by jjaferson** [comment]
> @jjaferson What's the reasoning here for not leveraging the Cloud Resource Operator here to create the bucket and provide the credentials, similar to #206 ?
> 
> Of course, we're still blocked on OSD, but the Cloud Resource Operator will create a placeholder secret at the moment until OSD has been unblocked.

Actually after looking at using the Cloud Resource Operator to automate the creation of s3 credentials and bucket, I am not sure it will be possible as this happens at bootstrap stage and CRO has not been reconciled yet

File: pkg/apis/integreatly/v1alpha1/installation_types.go [hunk 1]
```diff
@@ -91,12 +91,13 @@ type InstallationSpec struct {
...
-	NamespacePrefix  string         `json:"namespacePrefix,omitempty"`
-	SelfSignedCerts  bool           `json:"selfSignedCerts"`
-	PullSecret       PullSecretSpec `json:"pullSecret"`
+	Type                 string         `json:"type"`
+	RoutingSubdomain     string         `json:"routingSubdomain,omitempty"`
+	MasterURL            string         `json:"masterUrl,omitempty"`
+	NamespacePrefix      string         `json:"namespacePrefix,omitempty"`
+	SelfSignedCerts      bool           `json:"selfSignedCerts"`
+	PullSecret           PullSecretSpec `json:"pullSecret"`
+	UseExternalResources bool           `json:"useExternalResources"`
```
Comment (dimitraz) [inline pkg/apis/integreatly/v1alpha1/installation_types.go]: We've actually removed the use of this flag now, we provision resources with cro by default

File: pkg/controller/installation/products/cloudresources/reconciler.go [hunk 2]
```diff
@@ -86,10 +89,46 @@ func (r *Reconciler) Reconcile(ctx context.Context, inst *v1alpha1.Installation,
...
+
 	product.Host = r.Config.GetHost()
 	product.Version = r.Config.GetProductVersion()
 	product.OperatorVersion = r.Config.GetOperatorVersion()
 
 	r.logger.Infof("%s has reconciled successfully", r.Config.GetProductName())
 	return v1alpha1.PhaseCompleted, nil
 }
+
+func (r *Reconciler) reconcileS3BlobStorage(ctx context.Context, installation *v1alpha1.Installation, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
```
Comment (dimitraz) [inline pkg/controller/installation/products/cloudresources/reconciler.go]: Which product is this blob storage for, just out of curiosity?

File: pkg/apis/integreatly/v1alpha1/installation_types.go (hunk 1 above)
Comment (jjaferson) [reply pkg/apis/integreatly/v1alpha1/installation_types.go]: `UseExternalResources`, right? if so I will remove it

File: pkg/controller/installation/products/cloudresources/reconciler.go (hunk 2 above)
Comment (jjaferson) [reply pkg/controller/installation/products/cloudresources/reconciler.go]: it's for the integreatly operator, are the products now responsible for reconciling s3 credentials and bucket? if so I am not sure if we still need to have an blob storage only for the integreatly operator

File: pkg/apis/integreatly/v1alpha1/installation_types.go (hunk 1 above)
Comment (dimitraz) [reply pkg/apis/integreatly/v1alpha1/installation_types.go]: Yeah :+1:

File: pkg/controller/installation/products/cloudresources/reconciler.go (hunk 2 above)
Comment (dimitraz) [reply pkg/controller/installation/products/cloudresources/reconciler.go]: Yeah so the individual products would be responsible for creating the custom resources for the databases/caches that they need. The cloud resource operator picks up on the custom resource and provisions the db and the connection secret that the product will use to connect to the resource

Comment (jjaferson) [reply pkg/controller/installation/products/cloudresources/reconciler.go]: Great, so I don't think we need s3 credentials and bucket for the integreatly operator as each individual product creates their own resources. 

ping @philbrookes

**Pull request closed without merging**